packages/mobile/src/transactions/feed/queryHelper.ts

```diff
@@ -0,0 +1,269 @@
+    )
+
+    transactionsWithoutDuplicatedHash.sort((a, b) => {
+      return b.timestamp - a.timestamp
+    })
+
+    setFetchedTransactions(transactionsWithoutDuplicatedHash)
+  }
+}
+
+export function useFetchTransactions(): QueryHookResult {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const address = useSelector(walletAddressSelector)
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+
+  const [pageInfo, setPageInfo] = useState<PageInfo | null>(null)
+  const [fetchingMoreTransactions, setFetchingMoreTransactions] = useState(false)
+  const [transactions, setTransactions] = useState<TokenTransaction[]>([])
+
+  // Update the counter variable every |POLL_INTERVAL| so that a query is made to the backend.
+  const [counter, setCounter] = useState(0)
+  useInterval(() => setCounter((n) => n + 1), POLL_INTERVAL)
+
+  const addTransactions = addDeduplicatedTransactions(transactions, setTransactions)
+
+  const handleResult = (result: QueryResponse, paginatedResult: boolean) => {
+    Logger.info(TAG, `Fetched ${paginatedResult ? 'next page' : 'new'} transactions`)
+
+    const returnedTransactions = result.data?.tokenTransactionsV2?.transactions
+    if (returnedTransactions?.length) {
+      addTransactions(returnedTransactions)
+      // We store non-paginated results in redux
+      if (!paginatedResult) {
+        dispatch(updateTransactions(returnedTransactions))
+      }
+    }
+
+    const returnedPageInfo = result.data?.tokenTransactionsV2?.pageInfo
+    if ((!pageInfo || paginatedResult) && returnedPageInfo) {
+      setPageInfo(returnedPageInfo)
+    }
+  }
+
+  const handleError = (error: Error) => {
+    dispatch(showError(ErrorMessages.FETCH_FAILED))
+    Logger.error(TAG, 'Error while fetching transactions', error)
+  }
+
+  // Query for new transaction every POLL_INTERVAL
+  const { loading, error } = useAsync(
+    async () => {
+      const result = await queryTransactionsFeed(address, localCurrencyCode)
+      handleResult(result, false)
+    },
+    [counter],
+    {
+      onError: handleError,
+    }
+  )
+
+  // Query for next page of transaction if requested
+  useAsync(
+    async () => {
+      if (!fetchingMoreTransactions || !pageInfo?.hasNextPage) {
+        setFetchingMoreTransactions(false)
+        return
+      }
+
+      const result = await queryTransactionsFeed(address, localCurrencyCode, pageInfo?.endCursor)
+      setFetchingMoreTransactions(false)
+      handleResult(result, true)
+    },
+    [fetchingMoreTransactions],
+    {
+      onError: (e) => {
+        setFetchingMoreTransactions(false)
+        handleError(e)
+      },
+    }
+  )
+
+  return {
+    loading,
+    error,
+    transactions,
+    fetchingMoreTransactions,
+    fetchMoreTransactions: () => {
+      if (!pageInfo) {
+        dispatch(showError(ErrorMessages.FETCH_FAILED))
+      } else if (!pageInfo.hasNextPage) {
+        dispatch(showMessage(t('noMoreTransactions')))
```

Comment:
This will be called automatically, right? In that case, won't the "No more transactions" message always show up? I can imagine being a bit weird for users with very few transactions to see that message with almost no scrolling. Not a big deal in any case, I just wouldn't show any messages.

Reply:
oh yeah this is a good point...just checking, does this "no more transactions" message display as a banner like this? i'd have thought we want a small text at the bottom of the feed rather than a global banner (but i didn't test this, so i might be wrong!)

Reply:
The `showMessage` action does indeed show a blue banner at the top. It's quite easy to miss as well